src/r_util.cpp

```diff
@@ -473,6 +474,18 @@ namespace rhost {
             return R_NilValue;
         }
 
+        extern "C" SEXP get_file_lock_state(SEXP paths) {
+            R_len_t len  = Rf_length(paths);
+            std::vector<std::string> files;
+            
+            for (R_len_t i = 0; i < len; ++i) {
+                const char* file_path = R_CHAR(STRING_ELT(paths, i));
```

Comment:
This will get the string data in whatever encoding it was stored in the R character vector, which can vary. It will _usually_ be CP_ACP, but can also be Latin-1 or UTF-8. You need to call `Rf_translateCharUTF8`.

This will add another problem, in that `Rf_translateCharUTF8` can raise an R error, which will unwind the stack using `setjmp` (skipping any destructors). Meaning that you either have to avoid allocating any C++ objects on the stack before you call it, or you need to use `r_top_level_exec`.